src/utils.ts

```diff
@@ -241,3 +241,13 @@ const RE_RTL_CHECK = new RegExp(`^[^${RS_LTR_CHARS}]*[${RS_RTL_CHARS}]`);
 export const isRTL = (text: string) => {
   return RE_RTL_CHECK.test(text);
 };
+
+export const isTouchDevice = () => {
```

Comment:
We either listen for touchmove or mousemove events. Could you propagate the cursor type from there rather than trying to figure it out based on the window dimensions?

Reply:
@vjeux I was the one to suggest this workflow. If we do it your way we'll also need to track & reset the touch cursor once user potentially switches to mouse. While correct, it'll make the logic more complex, and I thought the `isTouchDevice` heuristic is fine for most cases. Furthermore, we're already depending on this heuristic for UI, and this would only make it consistent.

Reply:
I don’t understand why you need to “track and reset” anything. The system is designed to be stateless, we send all the info about cursor on every update. So if the user suddenly switches from mouse to touch, they’ll just send touch in their new stream of updates and it’ll just work out fine.

Reply:
Current implem isn't stateless. There's `state.remoteIsTouchDevice` (which is arguably necessary, same as `state.cursorButton` we use for up/down states) which would need to set/reset on `touchstart/pointerdown` (and it's unclear how we'd distinguish between mouse/touch in the `pointerdown` event if we don't do mobile detection).

Reply:
@dwelle do I proceed with current approach or do I change?

Reply:
If you can see a path forward along the way vjeux suggested and it isn't overly complicated, do that. Otherwise choose whatever you feel is best. Like I said below, I don't care either way --- I haven't investigated this much (at this point you definitely know more than I do).